cli/auth_tokens.rs

```diff
@@ -182,4 +260,69 @@ mod tests {
     let fixture = resolve_url("https://deno.land:8080/x/mod.ts").unwrap();
     assert_eq!(auth_tokens.get(&fixture), None);
   }
+
+  #[test]
+  fn test_parse_ip() {
+    let ip = AuthDomain::from("[2001:db8:a::123]");
+    assert_eq!("IP(2001:db8:a::123)", format!("{ip:?}"));
+    let ip = AuthDomain::from("[2001:db8:a::123]:8080");
+    assert_eq!("IPPort([2001:db8:a::123]:8080)", format!("{ip:?}"));
+    let ip = AuthDomain::from("1.1.1.1");
+    assert_eq!("IP(1.1.1.1)", format!("{ip:?}"));
+  }
+
+  #[test]
+  fn test_matches() {
+    let candidates = [
+      "example.com",
+      "www.example.com",
+      "notexample.com",
+      "www.notexample.com",
+      "1.1.1.1",
+      "[2001:db8:a::123]",
+    ];
```

Comment:
Can you add some tests with ports here?

Reply:
These are tested with ports -- we add a port to each candidate below.